python/cudf/cudf/core/column/column.py

```diff
@@ -2371,3 +2376,43 @@ def concat_columns(objs: "MutableSequence[ColumnBase]") -> ColumnBase:
                 ) from e
             raise
     return col
+
+
+def _create_list_column_from_sequences_list(arbitrary: List[ColumnLike]):
+    """
+    Create a list column for list of column-like sequences
+    """
+    data_col = as_column(arbitrary[0])
+    lengths_col = [len(data_col)]
+    mask_col = [True]
+
+    # Build Data & Mask
+    for data in arbitrary[1:]:
+        if cudf._lib.scalar._is_null_host_scalar(data):
+            mask_col.append(False)
+            lengths_col.append(0)
+        else:
+            mask_col.append(True)
+            data_col = data_col.append(as_column(data))
```

Comment:
Is it worth checking that the dtypes of each `arbitrary` element are consistent, or will a clear error be raised in the `ListColumn` constructor below?

Reply:
When we do `append`, we already check for this:
```python
>>> s = cudf.Series([1, 2, 3])
>>> cudf.Series([s, s, None, cudf.Series(['a'])])
Traceback (most recent call last):
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.8/site-packages/cudf/core/column/column.py", line 2032, in as_column
    memoryview(arbitrary), dtype=dtype, nan_as_null=nan_as_null
TypeError: memoryview: a bytes-like object is required, not 'list'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.8/site-packages/cudf/core/column/column.py", line 2080, in as_column
    pa.array(
  File "pyarrow/array.pxi", line 306, in pyarrow.lib.array
  File "pyarrow/array.pxi", line 39, in pyarrow.lib._sequence_to_array
  File "pyarrow/error.pxi", line 141, in pyarrow.lib.pyarrow_internal_check_status
  File "pyarrow/error.pxi", line 97, in pyarrow.lib.check_status
pyarrow.lib.ArrowInvalid: Could not convert 0    1
1    2
2    3
dtype: int64 with type Series: did not recognize Python value type when inferring an Arrow data type

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.8/site-packages/cudf/core/series.py", line 256, in __init__
    data = column.as_column(data, nan_as_null=nan_as_null, dtype=dtype)
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.8/site-packages/cudf/core/column/column.py", line 2108, in as_column
    return _create_list_column_from_sequences_list(arbitrary)
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.8/site-packages/cudf/core/column/column.py", line 2404, in _create_list_column_from_sequences_list
    data_col = data_col.append(as_column(data))
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.8/site-packages/cudf/core/column/column.py", line 680, in append
    return concat_columns([self, as_column(other)])
  File "/nvme/0/pgali/envs/cudfdev/lib/python3.8/site-packages/cudf/core/column/column.py", line 2346, in concat_columns
    raise ValueError("All columns must be the same type")
ValueError: All columns must be the same type
```